src/qililab/qprogram/qprogram.py

```diff
@@ -477,23 +513,34 @@ def set_offset(self, bus: str, offset_path0: float, offset_path1: float | None =
         self._active_block.append(operation)
         self._buses.add(bus)
 
+    @requires_domain("duration", Domain.Time)
+    def set_trigger(self, bus: str, duration: int, outputs: list[int] | int | None = None, position: str = "start"):
+        """Adds a delay on the bus and wait for an trigger signal to arrive.
+        Args:
+            bus (str): Unique identifier of the bus.
+            duration (int): Duration of the delay after the trigger is received. Minimum of 4 ns.
+        """
+        operation = SetTrigger(bus=bus, outputs=outputs, duration=duration, position=position)
+        self._active_block.append(operation)
+        self._buses.add(bus)
+
+    def set_markers(self, bus: str, mask: str):
+        """Set the markers based on a 4-bit binary mask.
+
+        Args:
+            bus (str): Unique identifier of the bus.
+            mask (str): A 4-bit mask, where 0 means that the associated marker is open (no signal), and 1 means that the marker is closed (signal).
+        """
+        operation = SetMarkers(bus=bus, mask=mask)
+        self._active_block.append(operation)
+        self._buses.add(bus)
```

Comment:
These are Qblox-only operations, right? Then we should add them in `_QbloxInterface`.